expertsystem/topology.py

```diff
@@ -0,0 +1,453 @@
+"""All modules related to topology building.
```

Comment:
Hmm, I think its better to rename this into graph. In my definition a topology would be just a graph without any properties attached to it. A general graph contains properties though, so graph would be the more general term.

Reply:
In a way, this touches on #196. For now I just followed the existing hierarchy.

Reply:
Lets also move that into a separate issue